codalab/apps/web/views.py

```diff
@@ -182,6 +181,37 @@ def get_object(self, queryset=None):
         return self.request.user
 
 
+class AdminCompetitionsManager(ModelFormSetView):
+    """Admin page for managing the competitions"""
+    model = Competition
+    template_name = "web/admin_competitions_manager.html"
+    fields = [
+        'id',
+        'title',
+        'creator',
+        'start_date',
+        'end_date',
+        'upper_bound_max_submission_size'
+    ]
+    success_url = '/admin_competitions_manager'
+    factory_kwargs = {'extra': 0}
+
+    def get(self, *args, **kwargs):
+        redirect_url = "index.html"
+        user = self.request.user
+        if user.is_staff and user.is_active:
+            return super(AdminCompetitionsManager, self).get(*args, **kwargs)
+        else:
+            return HttpResponseRedirect(redirect_url)
+
+    def get_context_data(self, **kwargs):
+        context = super(AdminCompetitionsManager, self).get_context_data(**kwargs)
+        competitions = models.Competition.objects.order_by('-pk')
```

Comment:
I don't remember off the top of my head, but is `start_date` not a field on the competition itself? (Necessitating sorted/reversed)?

Reply:
Yes start_date is a field on the Competition itself.
What do you mean by "_Necessitating sorted/reversed_" ?

Reply:
So if the `start_date` is not a field in the database. there would be no way to sort the initial query by that field (without some extra query magic).

> "Necessitating sorted/reversed"

Meaning since we cannot use queries to sort our data, sorted/reversed is used instead of a query:
```
list(reversed(sorted(competitions, key=lambda c: c.get_start_date)))
```

Reply:
Actually it seems that I can sort the competitions on the staret_date field using the query. I'm updating it